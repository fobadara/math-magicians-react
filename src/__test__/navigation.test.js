import { render, cleanup } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import '@testing-library/jest-dom/extend-expect';
import NavBar from '../components/navigation';

afterEach(cleanup);

const renderWithRouter = (component) => {
  const history = createMemoryHistory();
  return {
    ...render(
      <Router history={history}>
        {component}
      </Router>,
    ),
  };
};

describe('Navbar', () => {
  test('should create a snapshot', () => {
    const { asFragment } = renderWithRouter(<NavBar />);
    expect(asFragment(<NavBar />)).toMatchSnapshot();
  });
});
