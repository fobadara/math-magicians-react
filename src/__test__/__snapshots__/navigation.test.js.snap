// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should create a snapshot 1`] = `
<DocumentFragment>
  <div
    class="nav-border"
  >
    <nav
      class="navbar navbar-expand-lg navbar-light"
    >
      <div
        class="container"
      >
        <a
          class="navbar-brand"
          href="/"
        >
          Math Magicians
        </a>
        <button
          aria-controls="responsive-navbar-nav"
          aria-label="Toggle navigation"
          class="navbar-toggler collapsed"
          type="button"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
        <div
          class="navbar-collapse collapse"
          id="responsive-navbar-nav"
        >
          <div
            class="me-auto navbar-nav"
          />
          <div
            class="navbar-nav"
          >
            <a
              class="active nav-link"
              data-rr-ui-event-key="/"
              data-testid="home"
              href="/"
              id="/"
            >
              Home
            </a>
            <a
              class="nav-link"
              data-rr-ui-event-key="calculator"
              data-testid="quotes"
              href="/calculator"
              id="calculator"
            >
              Calculator
            </a>
            <a
              class="nav-link"
              data-rr-ui-event-key="quotes"
              data-testid="quotes"
              href="/quotes"
              id="quotes"
            >
              Quotes
            </a>
          </div>
        </div>
      </div>
    </nav>
  </div>
</DocumentFragment>
`;
